pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -146,14 +149,6 @@ func ShouldPodBeInEndpointSlice(pod *v1.Pod, includeTerminating bool) bool {
 		return false
 	}
 
-	if pod.Spec.RestartPolicy == v1.RestartPolicyNever {
```

Comment:
Do we know for certain this was working around the old bug?  I assume that we just didn’t catch this at the time, but I want to be certain there isn’t another bug.

Reply:
this one https://github.com/kubernetes/kubernetes/pull/47504???

Reply:
I meant the old bug that sometimes pods could be terminal and have running containers, vs some other bug which we might still have to fix.

Reply:
I think that endpoints should just assume pod terminal will never regress .
We should work now on adding more coverage to the pod lifecycle (once we have it clearly defined), so we never regress there,  @bowei commented today in the sig-net meeting that he'll be pushing this